commit 3c8387188a699c797bef08c8a24ad2f988cd994c
commit 6ca0e5dee4142d7995827b28ff674006ee3f8dd4

--- a/applets/clock/clock.c
+++ b/applets/clock/clock.c
@@ -1213,7 +1213,6 @@ location_set_current_cb (ClockLocation *loc,
 	if (cd->map_widget)
 		clock_map_refresh (CLOCK_MAP (cd->map_widget));
         update_location_tiles (cd);
-	save_cities_store (cd);	
 }
 
 static void
@@ -1617,8 +1617,7 @@ edit_clear (ClockData *cd)
         GtkWidget *lon_combo = _clock_get_widget (cd, "edit-location-longitude-combo");
 
         /* clear out the old data */
-        /* TODO: reinstate after bumping to gweather >= 3.10.1 */
-        //gweather_location_entry_set_location (cd->location_entry, NULL);
+        gweather_location_entry_set_location (cd->location_entry, NULL);
         gweather_timezone_menu_set_tzid (cd->zone_combo, NULL);
 
         gtk_entry_set_text (GTK_ENTRY (lat_entry), "");
--- a/configure.ac
+++ b/configure.ac
@@ -96,7 +96,7 @@ TELEPATHY_GLIB_REQUIRED=0.14.0
 LIBECAL_REQUIRED=3.5.3
 LIBEDATASERVER_REQUIRED=3.5.3
 CAIRO_REQUIRED=1.0.0
-GWEATHER_REQUIRED=3.9.2
+GWEATHER_REQUIRED=3.10.1
 DCONF_REQUIRED=0.13.4
 LIBRSVG_REQUIRED=2.36.2
 
